ironfish-rust/src/keys/mod.rs

```diff
@@ -210,28 +207,11 @@ impl SaplingKey {
         &self.incoming_viewing_key
     }
 
-    /// Retrieve the sapling_representation of the ViewKey. These would normally used for third-party audits
-    /// or for light clients.
-    /// Adapter to convert this key to a viewing key for use in sapling
-    /// functions.
-    pub(crate) fn sapling_viewing_key(&self) -> ViewingKey {
-        ViewingKey {
-            ak: self.authorizing_key,
-            nk: self.nullifier_deriving_key,
-        }
-    }
```

Comment:
Is there some reason we want to use getter pattern here, or can I just delete and access the struct directly? I noticed that the `outgoing` and `incoming` view keys have the pattern of getter method, but wondering what is idiomatic in rust

Reply:
I am just gonna leave this method for now since I don't want to diverge from the cannon of how the fields on the `SaplingKey` are approached, but I think this is a useful discussion.

Reply:
This sums up my feelings about getters: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=b78b2a1de22a603e84625616b49e3a33

If there's not a scenario where we would expect re-assignment, why even allow it? Accidents can happen. Yeah, we get some safety with mutable/immutable vars, but it's a lot easier to toss `mut` on something than to have to go change a whole function signature. In that vein, I want to go through and remove most of our `pub` keywords one of these days

Reply:
In this case, I'd say we're running into two slightly different patterns. Most of these previously were just functions to return a subset of the SaplingKey data via a more-specific struct. I'm not sure off the top of my head why IVK/OVK diverge from that pattern, and whether that makes sense or not. I'll defer to you on that for now